troncos/traces/decorate.py

```diff
@@ -99,6 +103,8 @@ def myfunc1()
     def myfunc2()
         return "This will be traced using a custom provider"
     """
+    if resource:
+        attributes["resource"] = resource
```

Comment:
Attributes can be None. Just pass `resource` into `_trace_function` below and be done with it.